### DISCUSSION TITLE
SvelteKit Docker fetch issue

### DISCUSSION THREAD

**Discussion by EfremovEgor** [sveltejs/kit#11890] [Q&A] [answered]
I have searched for many hours but haven't found any answer to this problem: I have 2 docker containers. One with FastAPI backend and one with Sveltekit app. 

Sending requests with openapi docs is ok. Pages are rendered fine. Sveltekit with npm run dev works fine too. But SvelteKit fetch (in +page.server.js) in docker container throws error (no error on local machine)

```
frontend  | TypeError: fetch failed
frontend  |     at node:internal/deps/undici/undici:12443:11
frontend  |     at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
frontend  |     at async resolve2 (file:///app/build/server/index.js:4118:14)
frontend  |     at async respond (file:///app/build/server/index.js:3941:22)
frontend  |     at async fetch (file:///app/build/server/index.js:3696:26) {
frontend  |   cause: Error: Client network socket disconnected before secure TLS connection was established   
frontend  |       at TLSSocket.onConnectEnd (node:_tls_wrap:1727:19)
frontend  |       at TLSSocket.emit (node:events:531:35)
frontend  |       at endReadableNT (node:internal/streams/readable:1696:12)
frontend  |       at process.processTicksAndRejections (node:internal/process/task_queues:82:21) {
frontend  |     code: 'ECONNRESET',
frontend  |     path: undefined,
frontend  |     host: '192.168.0.13',
frontend  |     port: 443,
frontend  |     localAddress: null
frontend  |   }
frontend  | }
```
## Vite config
```
export default defineConfig({
    plugins: [sveltekit()],

    server: {
        host: true,
        proxy: {
            '/api': {
                target: 'http://api:8000', // see here
                changeOrigin: true,
                secure: false,
                rewrite: (path) => path.replace(/^\/api/, '')
            }
        }
    }
});
```
Changing http://api:8000 to localhost:8000 / localhost had no effect.

I tried ```curl -XGET http://api:8000/conferences/ ``` from frontend docker container and it works just fine.
Also  ```process.env['NODE_TLS_REJECT_UNAUTHORIZED'] = 0;``` didn't work.

Creating additional express server and passing handler.js didn't work too.

Using nginx didn't help either

**Comment by wwwdepot** [comment, marked as answer]
> Configure custom proxy rules for the dev server.
> https://vitejs.dev/config/server-options.html#server-proxy

Vite `server.proxy` only works during dev. For production, see [How do I use a different backend API server?](https://kit.svelte.dev/docs/faq#how-do-i-use-x-with-sveltekit-how-do-i-use-a-different-backend-api-server)

**Comment by EfremovEgor** [comment]
Thanks for your response. I added this to hooks.server.js and it worked!

```
import { backend_url } from './utils';

export async function handleFetch({ request, fetch }) {
	request = new Request(`http://api:8000${request.url.split(backend_url)[1]}`, request);
	return fetch(request);
}
```
Where api is name of docker container
